src/Tools/ExternalAccess/Razor/RazorAnalyzerAssemblyResolver.cs

```diff
@@ -54,5 +55,12 @@ internal static bool TrySetAssemblyResolver(Func<AssemblyName, Assembly?> resolv
 
             return resolver?.Invoke(assemblyName);
         }
+
+        private class AssemblyNameEqualityComparer : IEqualityComparer<AssemblyName>
+        {
+            public bool Equals(AssemblyName? x, AssemblyName? y) => x?.FullName.Equals(y?.FullName) == true;
```

Comment:
It's not obvious to me this does the right thing when one or both AssemblyNames are null. Maybe it would be simpler to branch like `x is null ? y is null : x.FullName.Equals(y.FullName)`.

Reply:
On .NET core only the short name is considered for loading purposes in the majority case. This code is going to differentiate based on the full name though.

Reply:
Yeah, so I mentioned this on Aleksey's comment, but we control the parent interface which is annotated as never being null, and the comparer is only ever used here. I've updated it to `!` those values to indicate we know it will never be null as opposed to making the test stronger. Happy to switch back to a stronger null check if you prefer though.

Reply:
>On .NET core only the short name is considered for loading purposes in the majority case. This code is going to differentiate based on the full name though.

This was deliberate. Full name includes the version string i.e something like `Microsoft.CodeAnalysis.Razor.Compiler, Version=9.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e`

Whereas `.Name` (which I assume is what you mean by shortname?) only includes the assembly name itself (`Microsoft.CodeAnalysis.Razor.Compiler`). Given that this is AssemblyLoadContext code its completely valid to load two different versions of the same assembly.

If Razor specifically asks for version 9.0.0.0 I don't want to consider an 8.0.0.0 version to be equivalent.